src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -63,7 +63,7 @@ class PyTest(PythonToolBase):
     # This should be kept in sync with `requirements.txt`.
     # TODO: To fix this, we should allow using a `target_option` referring to a
     #  `python_requirement` to override the version.
-    default_version = "pytest>=6.2.4,<6.3"
+    default_version = "pytest>=6.2.4,<8"
```

Comment:
I think that if we're expecting / have-a-default-lock-of `7`, then we shouldn't mention `6` here, and should instead bump the minimum end of the range to the version that you ended up testing with.

While it's possible that folks using plugins while relying on this default value might need to actually set the version if their plugins are no longer compatible with `7`, that seems fine to me.

Reply:
why not?
This allows downstream repos that have pants plugins (and thus use `pantsbuild.pants.testutil `) use pytest 6.
I don't think keeping tight dependency between pantsbuild.pants.testutil  and pytest is a good idea when there is no technical reason I can think of to do so.

Reply:
> This allows downstream repos that have pants plugins (and thus use `pantsbuild.pants.testutil `) use pytest 6.

They can still do that, by explicitly setting the version. This is just the default.

Reply:
Hm, we could have our `requirements.txt` be loose, but have `pytest.py` be tighter. Note that `pytest.py`'s version only needs to be compatible with `requirements.txt`, they don't need to be identical. `pytest.py` is what is used for the default _tool lockfile_, whereas `requirements.txt` gets used by `python_distribution` for the `install_requires` of `pansbuild.pants.testutil`